lib/src/features/settings/pages/profile_settings_page.dart

```diff
@@ -1,27 +1,49 @@
+import 'package:fit_and_healthy/src/features/settings/metrics_controller.dart';
 import 'package:fit_and_healthy/src/features/settings/settings_controller.dart';
 import 'package:fit_and_healthy/src/nested_scaffold.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:fit_and_healthy/src/common/styles/sizes.dart';
+import 'package:fit_and_healthy/shared/models/gender.dart';
 
-final genderProvider = StateProvider<Gender>((ref) => Gender.male);
-final heightProvider = StateProvider<String>((ref) => '170 cm');
-
-enum Gender { male, female }
-
-class ProfileSettingsPage extends ConsumerWidget {
+class ProfileSettingsPage extends ConsumerStatefulWidget {
   const ProfileSettingsPage({super.key});
 
   static const route = '/profile';
   static const routeName = 'Profile';
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final settingsState = ref.watch(settingsControllerProvider).value;
+  _ProfileSettingsPageState createState() => _ProfileSettingsPageState();
+}
+
+class _ProfileSettingsPageState extends ConsumerState<ProfileSettingsPage> {
+  String _height = '170 cm'; // Default height
+  Gender _gender = Gender.male; // Default gender
+  DateTime? _birthday; // Null by default
+
+  @override
+  void initState() {
+    super.initState();
+    _loadUserData();
+  }
+
+  Future<void> _loadUserData() async {
+    final metricsController = ref.read(metricsControllerProvider.notifier);
 
-    final selectedGender = ref.watch(genderProvider);
-    final height = ref.watch(heightProvider);
+    final height = await metricsController.getHeight();
+    final gender = await metricsController.getGender();
+    final birthday = await metricsController.getBirthday();
+
+    setState(() {
```

Comment:
You should not store the same state multiple places. The state is already stored in the metricsController and if you use `.watch()`, should automatically update the UI from the controller state.